proposals/0002-gep.md

```diff
@@ -0,0 +1,52 @@
+---
+MEP: 0002
+Title: Unifed API
+Discussion: Link
+Implementation: Link
+---
+
+# TITLE: Unified API
+
+## Abstract
+
+This GEP discusses the unified API that handles the interaction with LLM providers.
+
+## Motivation
+
+[TBU, what problem we are trying to solve, focus on users, use cases, and all important context]
+
+The Unified API holds all the logic needed to create a request to a provider. This solution allows the clients to be easily
+envoked by a provider pool.
+
+
+### Requirements
+
+R1: User should be able to specify specific models or select auto mode (selection handled by Glide)
+R2: Should be a Hertz Client
+R3: Should take a user request as input
```

Comment:
Some of requirements I see:
- API schemas must unify common LLM request params (e.g. topP, N, etc)
- API routes must unify common LLM endpoints/API
- API schemas should allow to customize provider-specific params (e.g. a different messages for OpenAI/Anthropic)

Reply:
@mkrueger12 could you then pull these items on the list?